core/src/main/java/org/mule/runtime/core/internal/processor/strategy/ProactorStreamEmitterProcessingStrategyFactory.java

```diff
@@ -148,7 +151,7 @@ public Sink createSink(FlowConstruct flowConstruct, ReactiveProcessor function)
             currentThread().interrupt();
             throw new MuleRuntimeException(e);
           }
-        }, createOnEventConsumer(), bufferSize);
+        }, createOnEventConsumer(), sinkBufferSize);
         sinks.add(new ProactorSinkWrapper<>(sink));
```

Comment:
Try to use a `RoundRobinFluxSinkSupplier` for consistency with the policies code.

Reply:
The supplier works with `FluxSinks` directly, while in this case we need our own `Sink`.